core/src/main/java/org/apache/spark/memory/MemoryConsumer.java

```diff
@@ -116,13 +110,7 @@ public void freeArray(LongArray array) {
   protected MemoryBlock allocatePage(long required) {
     MemoryBlock page = taskMemoryManager.allocatePage(Math.max(pageSize, required), this);
     if (page == null || page.size() < required) {
```

Comment:
While here, pull this check into the method and call it "checkAllocation" or something?

Reply:
@srowen I didn't do this on purpose so that `allocatePage` compiles smaller since the condition isn't true very often. Not worth the duplication here?